### PR TITLE
Feature/nihss

### PR DESCRIPTION
**Pull request by Murmelmannen**
# Guidance

Another developer and a domain expert (doctor) should review and approve the Pull Request before merge.

### Checklist

- [x] I have added unit tests for happy paths
- [x] I have added unit tests for sad paths
- [x] I have updated the README
- [x] I have added relevant references to external sources in the PR description (such as authors and external links)
Källa: Lyden, 2017
- https://www.ahajournals.org/doi/10.1161/strokeaha.116.015434

### CODE REVIEW

File: src/__tests__/nihss.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+test('it should return 17', () => {
+    const fields = {
+        correctMonth: true,
+        correctAge: true,
+        correctEyes: false,
+        correctHand: false,
+    }
+    expect(calculateStrokeScale(1, fields, 0, 2, 0, 3, 4, 0, 0, 0, 2, 5, 0, 0)).toEqual(19)
+})
```
Comment (martify) [inline src/__tests__/nihss.ts:11]: As inspiration, maybe this is more declarative (test is easier to read). What do you think?

```
test('it should return 17', () => {
  const scaleItems: ScaleItems = {
    awake: 1,
    orientations: {
      correctMonth: true,
      correctAge: true,
      correctEyes: false,
      correctHand: false,
    },
    eyes: 0,
    sight: 2,
    face: 0,
    armRight: 3,
    armLeft: 4,
    legRight: 0,
    legLeft: 0,
    ataxi: 0,
    sensibility: 2,
    language: 5,
    dysarthria: 0,
    sensory: 0,
  }
  expect(calculateStrokeScale(scaleItems)).toEqual(19)
})
```

Consequently

```
type Orientation = {
  correctMonth: boolean
  correctAge: boolean
  correctEyes: boolean
  correctHand: boolean
}

export type ScaleItems = {
  awake: number
  orientations: Orientation
  eyes: number
  sight: number
  face: number
  armRight: number
  armLeft: number
  legRight: number
  legLeft: number
  ataxi: number
  sensibility: number
  language: number
  dysarthria: number
  sensory: number
}

function countOrientation(fields: Orientation): number {
  let score = 0
  Object.values(fields).forEach(value => {
    if (!value) {
      score++
    }
  })
  return score
}

function calculateStrokeScale({
  orientations,
  ...otherItems
}: ScaleItems): number {
  const sumOther = Object.values(otherItems).reduce((sum, x) => sum + x, 0)
  return countOrientation(orientations) + sumOther
}

export {calculateStrokeScale}
```

File: src/nihss.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+function countOrientation(fields: Orientation): number {
+    let score = 0
+    Object.values(fields).forEach(value => {
+        if(!value){
+            score++
+        }
+    })
+    return score
```
Comment (martify) [inline src/nihss.ts:15]: If you want to use `reduce`, something like this should also work

```suggestion
    return Object.values(orientations).reduce((sum, isSet) => (isSet ? sum : sum + 1), 0)
```

**Review by martify — APPROVED** [review]

File: src/__tests__/nihss.ts (hunk 1 above)
Comment (Murmelmannen) [reply src/__tests__/nihss.ts:11]: I'll keep it in mind for the future

File: src/nihss.ts (hunk 2 above)
Comment (Murmelmannen) [reply src/nihss.ts:15]: Thank you! That will help

**Pull request merged**